File: 2.py
M = 4000000

fib1, fib2, fib = 1, 2, 3
s = 2
while fib < M:
    if fib % 2 == 0:
        s += fib
    fib1 = fib2
    fib2 = fib
    fib = fib1 + fib2

print(s)

File: cpp version/2.cpp
#include <iostream>

using namespace std;

int main() {
    int M = 4000000;

    int fib1 = 1, fib2 = 2, fib = 3;
    int sum = 2;

    while (fib < M)
    {
        if (fib % 2 == 0)
        {
            sum += fib;
        }
        fib1 = fib2;
        fib2 = fib;
        fib = fib1 + fib2;
    }
    
    
    cout << sum << endl;
}